### PR TITLE
Jetty 9.4.x 2903 delay listener instantiation

### PR DESCRIPTION
**Pull request by janbartel**
Listener classes have been eagerly instantiated, however the instantiation should be delayed until  the class is going to be used. This avoids unnecessary load of the listener class during quickstart preparation.

### CODE REVIEW

File: jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java [hunk 1]
```diff
@@ -356,13 +356,16 @@ protected void startContext() throws Exception
...
         {
-            // Call decorators on all holders, and also on any EventListeners before
-            // decorators are called on any other classes (like servlets and filters)
+            //Ensure listener instances are created, added to ContextHandler
             if(_servletHandler.getListeners() != null)
             {
                 for (ListenerHolder holder:_servletHandler.getListeners())
                 {             
-                    _objFactory.decorate(holder.getListener());
+                    holder.start();
```
Comment (gregw) [inline jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java]: Are holders stopped and/or destroyed with the context?

Comment (gregw) [inline jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java]: Should holders be explicitly started or can they be added as beans to be managed?
Are they currently listed in a Dump?

Comment (janbartel) [reply jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java]: Holders are stopped when the ServletHandler is stopped (same as ServletHolders and FilterHolders). Listener instances are cleared out on the ContextHandler when the context stops.

Comment (janbartel) [reply jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java]: All holders (Servlet,Filter,Listener) are added as unmanaged beans, and listed as such in the dump output.

**Pull request merged**